pkg/gui/context/local_commits_context.go

```diff
@@ -39,9 +41,32 @@ func NewLocalCommitsContext(c *ContextCommon) *LocalCommitsContext {
 
 		showYouAreHereLabel := c.Model().WorkingTreeStateAtLastCommitRefresh == enums.REBASE_MODE_REBASING
 
+		currentBranchName := ""
+		if rebasedBranch := c.Model().BranchBeingRebased; rebasedBranch != "" {
```

Comment:
I've been working on worktrees lately and I've had a similar need to know what branch a worktree is on, even if it's rebasing or in the middle of a git bisect, because in those cases the branch is considered to be 'owned' by the worktree, even if `git worktree list` just reports it as having a detached head. For example, you can't check out a branch that's checked out by another worktree, even if that worktree is in a rebase operation with that branch.

So I think we should have a `c.Model().CheckedOutBranch` to capture all of the stuff that you've added here. That way, I'll be able to update that to support worktrees on my end. Given we're not using BranchBeingRebased anywhere else, we can remove that.

What do you think?

Reply:
Introducing `c.Model().CheckedOutBranch` makes sense to me, but I wasn't entirely sure how. Here's one version: c05719b702bf. In this case, I still set the member at the end of `refreshCommitsWithLimit()` like I did before with `BranchBeingRebased`. Not sure this is the best place to do that. Also, I can't use the other model members in the implementation like I did before (`BisectInfo` and `Branches`) because I can't be sure they have been set at this point, so I'm reading the information again here.

An alternative could have been to store `branchBeingRebased` in Model like I did before (but maybe as a private field), and then have an accessor function for `CheckedOutBranch()` that computes the result based on that and the other model fields, similar to what I did before in `local_commits_context.go`. It would be a bit unusual because Model doesn't have any methods right now.

Let me know what your preference is!